core/src/services/icloud/session.rs

```diff
@@ -0,0 +1,304 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::collections::{BTreeMap, HashMap};
+use std::fmt::{Debug, Formatter};
+
+use http::header;
+use http::Method;
+use http::Request;
+use http::Response;
+use http::StatusCode;
+
+use crate::types::Result;
+use crate::{Error, ErrorKind};
+
+use crate::raw::oio::WriteBuf;
+use crate::raw::{new_json_deserialize_error, AsyncBody, HttpClient, IncomingAsyncBody};
+use crate::services::icloud::error::parse_error;
+use crate::services::icloud::webservices::iCloudWebservicesResponse;
+use serde_json::json;
+
+//TODO It just support cn
+//Other country shoule be https://www.icloud.com
+//Need to improve?
+const GLOBAL_HEADERS: [(&str, &str); 2] = [
```

Comment:
It's better to make it a config if `Origin` and `Referer` required header for requesting